### PR TITLE
deepcopy stored on module for AK_ArrayDeepCopy

### PR DESCRIPTION
**Pull request by flexatone**

### CODE REVIEW

**Comment by flexatone** [comment]
This change improves performance from as low as .93% to at least 1.0.
```
cls                     func                    ak                      ref                     ref/ak                  
ArrayDeepcopy           memo_new                18.62973568             19.74299165             1.05975694              
ArrayDeepcopy           memo_shared             0.04034022              0.04228398              1.0481842             
```

File: src/_arraykit.c [hunk 1]
```diff
@@ -784,11 +779,26 @@ PyInit__arraykit(void)
 {
     import_array();
     PyObject *m = PyModule_Create(&arraykit_module);
+
+    PyObject *copy = PyImport_ImportModule("copy");
+    if (!copy) {
+        Py_DECREF(m);
```
Comment (brandtbucher) [inline src/_arraykit.c:785]:
```suggestion
        Py_XDECREF(m);
```

File: src/_arraykit.c [hunk 2]
```diff
@@ -784,11 +779,26 @@ PyInit__arraykit(void)
...
+        return NULL;
+    }
+    PyObject *deepcopy = PyObject_GetAttrString(copy, "deepcopy");
+    Py_DECREF(copy);
+    if (!deepcopy) {
+        Py_DECREF(m);
```
Comment (brandtbucher) [inline src/_arraykit.c:791]:
```suggestion
        Py_XDECREF(m);
```

**Review by brandtbucher — CHANGES_REQUESTED** [review]
Nice! Just two small fixes (we haven't checked `m` for `NULL` yet at this point):

File: src/_arraykit.c (hunk 1 above)
Comment (flexatone) [reply src/_arraykit.c:785]: Thanks!

File: src/_arraykit.c (hunk 2 above)
Comment (flexatone) [reply src/_arraykit.c:791]: 👍🏽

**Review by brandtbucher — APPROVED** [review]

**Pull request merged**